samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java

```diff
@@ -0,0 +1,123 @@
+import com.microsoft.recognizers.datatypes.timex.expression.Constants;
+import com.microsoft.recognizers.datatypes.timex.expression.TimexProperty;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ *
+ */
+public class BookingDialog extends CancelAndHelpDialog {
+    private final String destinationStepMsgText = "Where would you like to travel to?";
+    private final String originStepMsgText = "Where are you traveling from?";
+
+    /**
+     *
+     */
+    public BookingDialog() {
+        super(BookingDialog.class.getName());
+
+        addDialog(new TextPrompt(TextPrompt.class.getName()));
+        addDialog(new ConfirmPrompt(ConfirmPrompt.class.getName()));
+        addDialog(new DateResolverDialog(null));
+        WaterfallStep[] waterfallSteps = {
+            this::destinationStep,
+            this::originStep,
+            this::travelDateStep,
+            this::confirmStep,
+            this::finalStep
+        };
+        addDialog(new WaterfallDialog(WaterfallDialog.class.getName(), Arrays.asList(waterfallSteps)));
+
+        // The initial child Dialog to run.
+        setInitialDialogId(WaterfallDialog.class.getName());
+    }
+
+    private CompletableFuture<DialogTurnResult> destinationStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+
+        if (bookingDetails.getDestination() == null) {
+            Activity promptMessage =
+                MessageFactory.text(destinationStepMsgText, destinationStepMsgText, InputHints.EXPECTING_INPUT);
+            return stepContext.prompt(TextPrompt.class.getName(), new PromptOptions() {{
+                setPrompt(promptMessage);
+            }});
+        }
+
+        return stepContext.next(bookingDetails.getDestination());
+    }
+
+    private CompletableFuture<DialogTurnResult> originStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+
+        bookingDetails.setDestination(stepContext.getResult().toString());
+
+        if (StringUtils.isNotBlank(bookingDetails.getOrigin())) {
+            Activity promptMessage =
+                MessageFactory.text(originStepMsgText, originStepMsgText, InputHints.EXPECTING_INPUT);
+            return stepContext.prompt(TextPrompt.class.getName(), new PromptOptions() {{
+                setPrompt(promptMessage);
+            }});
+        }
+
+        return stepContext.next(bookingDetails.getOrigin());
+    }
+
+    private CompletableFuture<DialogTurnResult> travelDateStep(WaterfallStepContext stepContext) {
+        BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
+
+        bookingDetails.setTravelDate(stepContext.getResult().toString());
+
+        if (StringUtils.isNotBlank(bookingDetails.getTravelDate()) || isAmbiguous(bookingDetails.getTravelDate())) {
```

Comment:
```suggestion
        if (bookingDetails.getTravelDate() == null || isAmbiguous(bookingDetails.getTravelDate())) {
```

[C#](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/13.core-bot/Dialogs/BookingDialog.cs#L71) checks null